test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -962,6 +881,79 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			}
 		})
 
+		It("should be able to resolve DNS across windows and linux deployments", func() {
+			if eng.HasWindowsAgents() {
+				iisImage := "microsoft/iis:windowsservercore-1803" // BUG: This should be set based on the host OS version
+				windowsServerImage := "microsoft/windowsservercore:1803"
+
+				By("Creating a deployment running IIS")
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
+				windowsDeploymentName := fmt.Sprintf("iis-dns-%s-%v", cfg.Name, r.Intn(99999))
+				windowsIISDeployment, err := deployment.CreateWindowsDeploy(iisImage, windowsDeploymentName, "default", 80, -1)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Creating a nginx deployment")
+				nginxDeploymentName := fmt.Sprintf("nginx-dns-%s-%v", cfg.Name, r.Intn(99999))
+				linuxNginxDeploy, err := deployment.CreateLinuxDeploy("library/nginx:latest", nginxDeploymentName, "default", "")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Ensure there is a Running nginx pod")
+				running, err := pod.WaitOnReady(nginxDeploymentName, "default", 3, 30*time.Second, cfg.Timeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(running).To(Equal(true))
+
+				By("Ensure there is a Running iis pod")
+				running, err = pod.WaitOnReady(windowsDeploymentName, "default", 3, 30*time.Second, cfg.Timeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(running).To(Equal(true))
+
+				By("Exposing a internal service for the linux nginx deployment")
+				err = linuxNginxDeploy.Expose("ClusterIP", 80, 80)
+				Expect(err).NotTo(HaveOccurred())
+				linuxService, err := service.Get(nginxDeploymentName, "default")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Exposing a internal service for the windows iis deployment")
+				err = windowsIISDeployment.Expose("ClusterIP", 80, 80)
+				Expect(err).NotTo(HaveOccurred())
+				windowsService, err := service.Get(windowsDeploymentName, "default")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Connecting to Windows from another Windows deployment")
+				name := fmt.Sprintf("windows-2-windows-%s", cfg.Name)
+				command := fmt.Sprintf("iwr -UseBasicParsing -TimeoutSec 60 %s", windowsService.Metadata.Name)
```

Comment:
agree on using `Invoke-WebRequest`. It would be good to know if the network path fails too, which nslookup won't catch.

There are some known problems with DNS search suffices that were caught in Kubernetes conformance testing. We're still fixing those.